tests/e2e/e2e_egress_validator_test.go

```diff
@@ -45,6 +48,18 @@ var _ = OSMDescribe("Test Submit Egress Policy",
 				egressOut, err := palpha1.PolicyV1alpha1().Egresses(Td.OsmNamespace).Create(ctx, egressIn, v1.CreateOptions{})
 				Expect(err).ShouldNot(HaveOccurred())
 				Expect(egressOut).ShouldNot(BeNil())
+				//now test spec Matches. Should fail
+				derr = palpha1.PolicyV1alpha1().Egresses(Td.OsmNamespace).Delete(ctx, invalidTestName, v1.DeleteOptions{})
+				if derr != nil {
+					//don't error because name is not there first test but report
+					GinkgoWriter.Write([]byte("Error deleting last test service:\n" + derr.Error()))
+				}
+				egressIn.SetName(invalidTestName)
+				egressIn.Spec.Sources = []policy.SourceSpec{{Kind: validTestKind, Name: invalidTestName, Namespace: validTestName}}
+				egressIn.Spec.Matches = []corev1.TypedLocalObjectReference{{Kind: invalidTestKind}}
+				egressOut, err = palpha1.PolicyV1alpha1().Egresses(Td.OsmNamespace).Create(ctx, egressIn, v1.CreateOptions{})
+				Expect(err).Should(HaveOccurred())
+				Expect(egressOut).ShouldNot(BeNil())
 			})
```

Comment:
should this be Should?

Reply:
nope. this should have an error and the api should always return a value